src/util/side_metadata/global.rs

```diff
@@ -572,293 +681,53 @@ pub unsafe fn store(metadata_spec: SideMetadataSpec, data_addr: Address, metadat
     );
-    memory::zero(meta_start, meta_size);
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use crate::util::constants;
-    use crate::util::heap::layout::vm_layout_constants;
-    use crate::util::side_metadata::helpers;
-    use crate::util::test_util::serial_test;
-
-    #[test]
-    fn test_side_metadata_try_mmap_metadata_chunk() {
-        serial_test(|| {
-            let gspec = SideMetadataSpec {
-                scope: SideMetadataScope::Global,
-                offset: 0,
-                log_num_of_bits: 0,
-                log_min_obj_size: 0,
-            };
-            let lspec = SideMetadataSpec {
-                scope: SideMetadataScope::PolicySpecific,
-                offset: 0,
-                log_num_of_bits: 1,
-                log_min_obj_size: 0,
-            };
-
-            assert!(try_map_metadata_space(
-                vm_layout_constants::HEAP_START,
-                1,
-                helpers::meta_bytes_per_chunk(0, 0),
-                helpers::meta_bytes_per_chunk(1, 1)
-            ));
-
-            ensure_metadata_chunk_is_mmaped(gspec, vm_layout_constants::HEAP_START);
-            ensure_metadata_chunk_is_mmaped(lspec, vm_layout_constants::HEAP_START);
-            ensure_metadata_chunk_is_mmaped(
-                gspec,
-                vm_layout_constants::HEAP_START + vm_layout_constants::BYTES_IN_CHUNK - 1,
-            );
-            ensure_metadata_chunk_is_mmaped(
-                lspec,
-                vm_layout_constants::HEAP_START + vm_layout_constants::BYTES_IN_CHUNK - 1,
-            );
-
-            assert!(try_map_metadata_space(
-                vm_layout_constants::HEAP_START + vm_layout_constants::BYTES_IN_CHUNK,
-                vm_layout_constants::BYTES_IN_CHUNK + 1,
-                helpers::meta_bytes_per_chunk(3, 2),
-                helpers::meta_bytes_per_chunk(4, 2)
-            ));
-
-            ensure_metadata_chunk_is_mmaped(
-                gspec,
-                vm_layout_constants::HEAP_START + vm_layout_constants::BYTES_IN_CHUNK,
-            );
-            ensure_metadata_chunk_is_mmaped(
-                lspec,
-                vm_layout_constants::HEAP_START + vm_layout_constants::BYTES_IN_CHUNK,
-            );
-            ensure_metadata_chunk_is_mmaped(
-                gspec,
-                vm_layout_constants::HEAP_START + vm_layout_constants::BYTES_IN_CHUNK * 3 - 1,
-            );
-            ensure_metadata_chunk_is_mmaped(
-                lspec,
-                vm_layout_constants::HEAP_START + vm_layout_constants::BYTES_IN_CHUNK * 3 - 1,
-            );
 
-            ensure_munmap_metadata_chunk(
-                vm_layout_constants::HEAP_START,
-                helpers::meta_bytes_per_chunk(0, 0),
-                helpers::meta_bytes_per_chunk(1, 1),
+    let meta_start = address_to_meta_address(metadata_spec, start);
+    if cfg!(target_pointer_width = "64") || metadata_spec.scope.is_global() {
+        memory::zero(
```

Comment:
We should have code like this for the mmap code - some general mmap functions in `util::memory`, and use them in side metadata code.